src/marshmallow_sqlalchemy/schema.py

```diff
@@ -181,6 +181,16 @@ def get_instance(self, data):
             return self.session.query(self.opts.model).filter_by(**filters).first()
         return None
 
+    @ma.pre_load
+    def set_field_session(self, data, **kwargs):
+        """Pass session to nested fields."""
+        nested_fields = {
+            k: v for k, v in self.fields.items() if isinstance(v, ma.fields.Nested)
+        }
+        for field in nested_fields.values():
+            field.schema.session = self._session
```

Comment:
We should probably only set `session` if the schema is a `ModelSchema`.

Reply:
Have made this part of the new solution.